pages/confirmation.php

```diff
@@ -10,84 +10,81 @@
+	if ( ! empty( $confirmation_invoice ) ) {
+		// Show the invoice, but make sure we don't show $pmpro_msg again.
 		$pmpro_msg = false;
 		$pmpro_msgt = false;
-
-		// Include invoice.php.
 		echo pmpro_loadTemplate( 'invoice' );
+	} else {
+		?>
+		<ul>
+			<li><strong><?php esc_html_e('Account', 'paid-memberships-pro' );?>:</strong> <?php echo esc_html( $current_user->display_name );?> (<?php echo esc_html( $current_user->user_email );?>)</li>
+			<li><strong><?php esc_html_e('Membership Level', 'paid-memberships-pro' );?>:</strong> <?php if(!empty($current_user->membership_level)) echo esc_html( $current_user->membership_level->name ); else esc_html_e("Pending", 'paid-memberships-pro' );?></li>
+			<?php if( !empty( $current_user->membership_level->expiration_period ) && $current_user->membership_level->expiration_period == 'Hour' && apply_filters( 'pmpro_confirmation_display_hour_expiration', true, $current_user ) ){ ?>
+			<li><strong><?php esc_html_e('Expires In', 'paid-memberships-pro' );?>:</strong> <?php echo esc_html( $current_user->membership_level->expiration_number . ' ' . pmpro_translate_billing_period( $current_user->membership_level->expiration_period, $current_user->membership_level->expiration_number ) ); ?></li>
+			<?php }
+			?>
+		</ul>
+		<?php
 	}
-	else
-	{
-		$confirmation_message .= "<p>" . sprintf(__('Below are details about your membership account. A welcome email has been sent to %s.', 'paid-memberships-pro' ), $current_user->user_email) . "</p>";
 
-		/**
-		 * All devs to filter the confirmation message.
-		 * Documented above.
-		 * We also have a function in includes/filters.php that applies the the_content filters to this message.
-		 */
-		$confirmation_message = apply_filters("pmpro_confirmation_message", $confirmation_message, false);
-
-		echo wp_kses_post( $confirmation_message );
 	?>
-	<ul>
-		<li><strong><?php esc_html_e('Account', 'paid-memberships-pro' );?>:</strong> <?php echo esc_html( $current_user->display_name );?> (<?php echo esc_html( $current_user->user_email );?>)</li>
-		<li><strong><?php esc_html_e('Membership Level', 'paid-memberships-pro' );?>:</strong> <?php if(!empty($current_user->membership_level)) echo esc_html( $current_user->membership_level->name ); else esc_html_e("Pending", 'paid-memberships-pro' );?></li>
-		<?php if( !empty( $current_user->membership_level->expiration_period ) && $current_user->membership_level->expiration_period == 'Hour' && apply_filters( 'pmpro_confirmation_display_hour_expiration', true, $current_user ) ){ ?>
-		<li><strong><?php esc_html_e('Expires In', 'paid-memberships-pro' );?>:</strong> <?php echo esc_html( $current_user->membership_level->expiration_number . ' ' . pmpro_translate_billing_period( $current_user->membership_level->expiration_period, $current_user->membership_level->expiration_number ) ); ?></li>
-		<?php }
+	<p class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_actions_nav' ) ); ?>">
```

Comment:
This empty <p> is still showing when there is no content to show (I have a membership and an invoice). So I think this need a bit more architecting. Perhaps we could move the message regarding inactive membership level to its own paragraph. We don't need to use the actions nav p for this.

<?php if ( ! empty( $current_user->membership_level ) && empty( $confirmation_invoice ) ) { ?>
<p class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_actions_nav' ) ); ?>">
<span class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_actions_nav-right' ) ); ?>"><a href="<?php echo esc_url( pmpro_url( 'account' ) ); ?>"><?php esc_html_e( 'View Your Membership Account &rarr;', 'paid-memberships-pro' ); ?></a></span>
</p>
<?php } ?>

<?php if ( empty( $current_user->membership_level ) ) { ?>
<p class="<?php echo esc_attr( pmpro_get_element_class( 'pmpro_membership_pending' ) ); ?>"><?php esc_html_e( 'If your account is not activated within a few minutes, please contact the site owner.', 'paid-memberships-pro' ); ?></p>
<?php } ?>

Something like this... ?

Reply:
Agreed, I was just trying to keep the code functionality as close to what was already here. So should we get rid fo that `<p>` tag around the `if` statement and instead wrap the content within the `if`s in paragraph tags?